jscomp/core/js_dump.ml

```diff
@@ -97,43 +97,29 @@ type cxt = Ext_pp_scope.t
 let semi f = P.string f L.semi
 let comma f = P.string f L.comma
 
-let new_error name cause =
-  E.new_ (E.js_global Js_dump_lit.error) [ name; cause ]
-
-let exn_block_as_obj ~(stack : bool) (el : J.expression list) (ext : J.tag_info)
+let exn_block_as_obj ~(is_exception : bool) (el : J.expression list) (ext : J.tag_info)
     : J.expression =
   let field_name =
     match ext with
-    | Blk_extension -> (
+    | Blk_extension _ -> (
         fun i ->
           match i with 0 -> Literals.exception_id | i -> "_" ^ string_of_int i)
```

Comment:
This change should make extension fields have the same logic as variants. Currently if an extension has a payload, it starts with `_1` while normal variants start with `_0`. I think it makes sense to have it unified. Should I include the change in a separate PR?
```suggestion
          match i with 0 -> Literals.exception_id | i -> "_" ^ string_of_int (i - 1))
```